src/NuGet.Licenses/Controllers/LicenseController.cs

```diff
@@ -149,9 +150,64 @@ private ActionResult DisplayLicenseFromFile(string identifier, bool isException)
         private ActionResult DisplayCompositeLicenseExpression(LicenseOperator licenseExpressionRoot, string licenseExpression)
         {
             var meaningfulSegments = _licenseExpressionSegmentator.GetLicenseExpressionSegments(licenseExpressionRoot);
+
             var allSegments = _licenseExpressionSegmentator.SplitFullExpression(licenseExpression, meaningfulSegments);
 
+            var stopwatch = Stopwatch.StartNew();
+            foreach (var segment in allSegments.Where(s => IsLicenseOrException(s)))
+            {
+                var isValidSegment = false;
+                if (NuGetLicenseData.ExceptionList.TryGetValue(segment.Value, out var exceptionData))
+                {
+                    isValidSegment = true;
+                }
+                else
+                {
+                    try
+                    {
+                        var license = NuGetLicenseExpression.Parse(segment.Value) as NuGetLicense;
+                        if (license != null && license.IsStandardLicense)
+                        {
+                            isValidSegment = true;
+                        }
+                    }
+                    catch (NuGetLicenseExpressionParsingException e)
+                    {
+                        stopwatch.Stop();
+                        _logger.LogError(0, e, "Got exception while attempting to parse license expression {LicenseExpression}", segment.Value);
+                        return InvalidRequest($"Unable to parse the license expression: {segment.Value}");
+                    }
+                }
+
+                if (!isValidSegment)
+                {
+                    stopwatch.Stop();
+                    _logger.LogInformation("Validating composite licenseExpression uses: {elapsedTime} s", stopwatch.Elapsed);
+                    return UnknownLicense(segment.Value);
+                }
+
+                try
+                {
+                    if (!_licenseFileService.DoesLicenseFileExist(segment.Value))
```

Comment:
>DoesLicenseFileExist [](start = 45, length = 20)

When will we have a valid segment that doesn't exist in the `licenseFileService`?

Reply:
We don't have all the SPDX licenses' content in the App_Data. So it may happen when the customer tries to fetch the content of a SPDX license expression but we don't support.

Reply:
If that's the case, should the `InvalidError` provide a better error message?